source/agora/common/Serializer.d

```diff
@@ -151,3 +164,65 @@ public void serializePart (scope const(ubyte)[] record, scope SerializeDg dg)
     serializePart(record.length, dg);
     dg(record);
 }
+
+/*******************************************************************************
+
+    Unsigned Integers to Serialized variable length integer
+    and return it as a ubyte[].
+
+    VarInt Size
+    size <= 0xFC(253)  -- 1 byte   ubyte
+    size <= USHORT_MAX -- 3 bytes  (0xFD + ushort)
+    size <= UINT_MAX   -- 5 bytes  (0xFE + uint)
+    size <= ULONG_MAX  -- 9 bytes  (0xFF + ulong)
+
+    Params:
+        T = Type of unsigned integer to serialize
+        var = Instance of `T` to serialize
+
+    Returns:
+        The serialized convert variable length integer
+
+*******************************************************************************/
+
+public ubyte[] toVarInt (T) (const T var)
```

Comment:
This function allocates with the GC where it shouldn't. Use `dg` instead of appending to an array.

Reply:
Used dg.